core/store/migrate/migrations/0235_add_workflow_models.sql

```diff
@@ -0,0 +1,43 @@
+-- +goose Up
+-- +goose StatementBegin
+CREATE TYPE workflow_status AS ENUM (
+	'started',
+	'errored',
+	'timeout',
+	'completed'
+);
+
+CREATE TABLE workflow_executions (
+	id varchar(64) PRIMARY KEY,
+	workflow_id varchar(64) NOT NULL,
+	status workflow_status NOT NULL,
```

Comment:
Not for the workflow execution status; atm setting this is totally managed by the workflow engine, which can set the status differently at different times (eg. we set the status to timeout depending on a configuration value). I don't want to push this into the data layer since that will split application logic between the application and the db.

For fields that are mostly used for tracking (eg. timestamps) I have less of an issue with that.